src/binding/encoding.rs

```diff
@@ -43,24 +42,20 @@ pub fn utf8_encoding() -> Value {
 }
 
 pub fn enc_get_index(s: Value) -> EncodingIndex {
-    let idx = unsafe { encoding::rb_enc_get_index(s) };
-
-    idx
+    unsafe { encoding::rb_enc_get_index(s) }
 }
 
 pub fn find_encoding_index(name: &str) -> EncodingIndex {
     let cstr = CString::new(name).unwrap();
-    let idx = unsafe { encoding::rb_enc_find_index(cstr.as_ptr()) };
-
-    idx
+    unsafe { encoding::rb_enc_find_index(cstr.as_ptr()) }
 }
 
 pub fn encode(str: Value, to: Value, ecflags: c_int, ecopts: Value) -> Value {
     unsafe { encoding::rb_str_encode(str, to, ecflags, ecopts) }
 }
 
-pub fn econv_prepare_opts(opthash: Value, opts: *const Value) -> c_int {
-    unsafe { encoding::rb_econv_prepare_opts(opthash, opts) }
+pub fn econv_prepare_opts(opthash: Value, opts: *mut Value) -> c_int {
+    unsafe { encoding::rb_econv_prepare_opts(opthash, opts as *mut _) }
 }
```

Comment:
I'm curious about various changes throughout the code where we're going from `*const` to `*mut`.  Perhaps a simple explanation on why `Value` can be `*mut` would suffice.  Just a reply here should be fine.